tls/test_tls_alpn_basic.py

```diff
@@ -0,0 +1,176 @@
+__author__ = "Tempesta Technologies, Inc."
+__copyright__ = "Copyright (C) 2024 Tempesta Technologies, Inc."
+__license__ = "GPL2"
+
+import socket
+import ssl
+
+from framework import tester
+from framework.parameterize import param, parameterize, parameterize_class
+from helpers import dmesg, tf_cfg
+
+
+class TestALPN(tester.TempestaTest):
+    tempesta_template = {
+        "config": """
+        listen 443 proto=%(tempesta_proto)s;
+
+        tls_match_any_server_name;
+
+        srv_group default {
+            server %(server_ip)s:8000;
+        }
+
+        vhost tempesta-tech.com {
+           tls_certificate %(work_dir)s/tempesta.crt;
+           tls_certificate_key %(work_dir)s/tempesta.key;
+           proxy_pass default;
+        }
+        """
+    }
+
+    clients = [
+        {
+            "id": "deproxy",
+            "type": "deproxy",
+            "addr": "${tempesta_ip}",
+            "port": "443",
+            "ssl": True,
+            "ssl_hostname": "tempesta-tech.com",
+        },
+    ]
+
+    backends = [
+        {
+            "id": "deproxy",
+            "type": "deproxy",
+            "port": "8000",
+            "response": "static",
+            "response_content": "HTTP/1.1 200 OK\r\n"
+            "Content-Length: 10\r\n"
+            "Connection: keep-alive\r\n\r\n"
+            "0123456789",
+        }
+    ]
+
+    def init_config(self, proto):
+        self.get_tempesta().config.set_defconfig(
+            self.tempesta_template["config"]
+            % {
+                "tempesta_proto": proto,
+                "work_dir": tf_cfg.cfg.get("General", "workdir"),
+                "server_ip": tf_cfg.cfg.get("Server", "ip"),
+            }
+        )
+
+    @parameterize.expand(
+        [
+            param(
+                name="order_https",
+                tempesta_proto="https",
+                protocols=["http/1.1", "h2"],
+            ),
+            param(
+                name="order_h2",
+                tempesta_proto="h2",
+                protocols=["h2", "http/1.1"],
+            ),
+            param(
+                name="mixed_https",
+                tempesta_proto="https,h2",
+                protocols=["http/1.1"],
+            ),
+            param(
+                name="mixed_h2",
+                tempesta_proto="https,h2",
+                protocols=["h2"],
+            ),
```

Comment:
I think we still need to add a one case 
```python3
            param(
                name="mixed_h2",
                tempesta_proto="https,h2",
                protocols=["h2", "http/1.1"],
            ),
```
Which proto will be chosen in this case?